### PR TITLE
Feat/mutation

### PR DESCRIPTION
**Pull request by alinauroz**
Adds mutation for creating a subscriber

### CODE REVIEW

File: graphql/types/subscriber.ts [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+import prisma from '../../lib/prisma';
+import { builder } from '../builder';
+
+export const CreateSubscriberInput = builder.inputType('CreateSubscriberInput', {
+  fields: (t) => ({
+    email: t.string(),
+    language: t.string(),
+    country: t.string(),
+  })
+});
```
Comment (alinauroz) [inline graphql/types/subscriber.ts:10]: This is the input type of addSubscriber mutation. It accepts three fields i-e email, language and country.

File: graphql/types/subscriber.ts [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
 
 builder.prismaObject('Subscriber', {
   name: 'Subscriber',
   fields: (t) => ({
     id: t.exposeID('id'),
     email: t.exposeString('email'),
+    country: t.exposeString('country'),
+    language: t.exposeString('language'),
     // createdAt: t.expose('createdAt', { type: 'Date' }),
```
Comment (alinauroz) [inline graphql/types/subscriber.ts:19]: Subscriber did not have country and language in its schema but it was present in prisma schema. So I also added it here

File: graphql/types/subscriber.ts [hunk 3]
```diff
@@ -17,3 +27,20 @@ builder.queryField('subscribers', (t) =>
       prisma.subscriber.findMany({ ...query }),
   })
 )
+
+builder.mutationField('addSubscriber', (t) => 
+  t.prismaField({
+    type: 'Subscriber',
```
Comment (alinauroz) [inline graphql/types/subscriber.ts:33]: This is the mutation to add subscriber. Its type is Subscriber which means it returns an object of type Subscriber.

File: graphql/types/subscriber.ts [hunk 4]
```diff
@@ -17,3 +27,20 @@ builder.queryField('subscribers', (t) =>
...
+    args: {
+      input: t.arg({
+        type: CreateSubscriberInput,
+        required: true
+      })
+    },
```
Comment (alinauroz) [inline graphql/types/subscriber.ts:39]: Here the input arguments of this mutation are mentioned. CreateSubscriberInput is defined above.

File: graphql/types/subscriber.ts [hunk 5]
```diff
@@ -17,3 +27,20 @@ builder.queryField('subscribers', (t) =>
...
+    resolve: async (query, _parent, _args, _ctx, _info) => {
+      try {
+        return prisma.subscriber.create({ data: _args.input });
+      }catch (err) {}
+    }
+  })
+);
```
Comment (alinauroz) [inline graphql/types/subscriber.ts:46]: This is the resolver of this mutation. When this mutation is called it created a record is database.

File: graphql/types/user.ts [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import { builder } from '../builder'
 
-builder.prismaObject('Subscriber', {
```
Comment (alinauroz) [inline graphql/types/user.ts:3]: Subscriber object was defined twice so I commented one.

File: graphql/builder.ts [hunk 7]
```diff
@@ -1,7 +1,6 @@
 import SchemaBuilder from '@pothos/core'
 import PrismaPlugin from '@pothos/plugin-prisma'
 import type PrismaTypes from '@pothos/plugin-prisma/generated'
-import { DateResolver } from 'graphql-scalars'
```
Comment (Yedidya10) [inline graphql/builder.ts:4]: It is important that the createdAt and updatedAt are recorded in a database. Please add them to the mutation process.

**Review by Yedidya10 — CHANGES_REQUESTED** [review]
Please see the comment I wrote.

File: graphql/types/subscriber.ts (hunk 1 above)
Comment (Yedidya10) [reply graphql/types/subscriber.ts:11]: I also referred to the following comment in the previous commit:
It is important that the createdAt and updatedAt are recorded in a database. Please add them to the mutation process.

**Review by Yedidya10 — CHANGES_REQUESTED** [review]
I also referred to the following comment in the previous commit

**Review by Yedidya10 — APPROVED** [review]
It looks okay

**Review by Yedidya10 — APPROVED** [review]
it looks good

**Review by Yedidya10 — APPROVED** [review]
it looks good

File: graphql/types/subscriber.ts (hunk 1 above)
Comment (alinauroz) [reply graphql/types/subscriber.ts:11]: createdAt and updatedAt are recorded. There default values are set in the model in prisma schema.

**Pull request merged**